feat: include systemd service file in repo

rbjorklin opened the pull request:
Include systemd service file in repo to allow templating.

rbjorklin committed:
feat: include systemd service file in repo

vutny commented on nomad/defaults.yaml:
```diff
@@ -4,9 +4,8 @@
-  config_dir: '/etc/nomad'
-  version: '0.8.7'
-  service_hash: fdc512a9e7e6a55303f12d4405676ec7c8116af6
+  config_dir: '/etc/nomad.d'
```
I think you'd need to update the `pillar.example` file as well.

rbjorklin committed:
fix: update pillar.example

vutny reviewed (changes requested):
I haven't used this formula and Nomad in particular, but the changes look good to me.
One last thing we need to address is fixing paths in `test_nomad_config.py` for CI to pass successfully.

rbjorklin committed:
fix(tests): update nomad config folder path in tests

rbjorklin committed:
chore: update pillar.example to trigger tests

rbjorklin commented:
It seems Travis fails to verify that the service is in fact running. Any pointers on how to read the output from `systemd status nomad` within the tests would be appreciated.

vutny commented on pillar.example:
```diff
@@ -2,39 +2,27 @@
+        retry_join:
+        {% for _, addrs in salt.saltutil.runner('mine.get', tgt='*', fun='network.ip_addrs') | dictsort() %}
+          - {{ addrs[0] }}
+        {% endfor %}
```
@rbjorklin I think this addition is preventing Nomad agent to start.

The function must be allowed to call from Salt "mine" explicitly, but the tests seems do not do that.
So, when there are nothing rendered within the for loop, the `retry_join` value would be `None`, which is invalid I suppose.
I suggest you to check if there any values come out from the `mine.get` call first, and only then iterate over them to render complete `server_join` dictionary.

rbjorklin committed:
fix: comment retry_join in pillar example

The Travis CI testing breaks with the retry_join in there.
Leaving it in but commented as the example is still useful.

rbjorklin committed:
fix: bump version to re-trigger tests

rbjorklin commented:
Long abandoned. You should be using Kubernetes.
